src/circuit/mod.rs

```diff
@@ -1301,9 +1306,27 @@ mod test {
         (verification_key, proof)
     }
 
+    fn check_if_crs_exists_and_download(n: usize) {
+        let file_path = &format!("setup_2^{}.key", n);
+        let file_exists = std::path::Path::new(file_path).exists();
```

Comment:
I would be consistent in the notation. So if you `use std:fs:File` I would do the same with Path, i.e. `use std::path::Path;`

Reply:
I see that in the rest of the code, this is used. I prefer what I proposed above, but not sure what is more idiomatic.